portfolio/src/main/webapp/script.js

```diff
@@ -158,45 +158,62 @@ function deleteComment(comment) {
 }
 
 function getData() {
-  var xhr = new XMLHttpRequest();                 // Create XMLHttpRequest object
-
-  xhr.onload = function() {                       // When response has loaded
-    // The following conditional check will not work locally - only on a server
+  var xhr = new XMLHttpRequest();                 
+  xhr.onload = function() {                       
     if(xhr.status === 200) {
-      var text = xhr.responseText;                   // If server status was ok
-      refreshComments(JSON.parse(text)); // Update
+      var text = xhr.responseText;                  
+      refreshComments(JSON.parse(text)); 
     }
   };
-
-  xhr.open('GET', '/data', true);        // Prepare the request
-  xhr.send(null);                                 // Send the request
+  xhr.open('GET', '/data', true);        
+  xhr.send(null);                               
 }
 
 // Refreshes the data each 2 seconds
-setInterval(getData, 1000*2);
+const milsec = 2;
+setInterval(getData, 1000*milsec);
 
 // New comments(=comments) are added to the page, and old comments(=currentComments) are removed
 function refreshComments(comments){
-  for(var i = 0; i < currentComments.length; i++){
-    if(objNotInList(currentComments[i],comments)){
-      removeCommentElement(i);
-      currentComments.splice(i,1);
+  var i = comments.length - 1;
+  var j = currentComments.length - 1;
+  while(currentComments[0].timestamp >= comments[i].timestamp){
```

Comment:
What happens, for example, when the last comment is deleted? I am a bit worried about the termination condition for the loop.

Reply:
Well if we have: (a3> a2>a1) timestamps
comments = { a2 , a1}
Cc = {a3, a2, a1}
for i =  1 and j = 2 (we are going to have a1=a1)
for i = 0 and j =1 (a2=a2)
for i = 0 and j = 0 (we remove the visual element, but also remove a3 from the list)
for i = 0 and j = 0 ( we have a2 = a2 -> it enters on the branch with the break of the if)
It is a bit shady, but my argument is that the algorithm leads to case i = j = 0. If the comments are different removeElement() and if they are the same than break;

Reply:
What do you think?

Reply:
I will try to think of something better tho (less not-clear, bcs it's not very clear how it works for me neither)